Serial suite is timing out

<!-- Please only use this template for submitting reports about failing tests in Kubernetes CI jobs -->

**Which jobs are failing**:
https://k8s-testgrid.appspot.com/sig-release-master-blocking#gce-cos-master-serial

**Which test(s) are failing**:
should unmount if pod is force deleted while kubelet is down [Disruptive][Slow]

**Since when has it been failing**:
11/07/18 15:02

**Testgrid link**:
https://k8s-testgrid.appspot.com/sig-release-master-blocking#gce-cos-master-serial

**Reason for failure**:
The test case stops and starts kubelet, and it seems like kubelet is unable to start again

**Anything else we need to know**:

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind failing-test

It looks like kubelet keeps restarting.  The error right before restarting is:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/4323/artifacts/bootstrap-e2e-minion-group-tjh1/kubelet.log
```
F1108 07:23:17.005364   26737 kubelet.go:1375] failed to start Plugin Watcher. err: failed to traverse plugin socket path, err: error accessing path: /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-28f57e30-e2e2-11e8-b368-42010a800002/globalmount error: readdirent: input/output error
panic: send on closed channel
goroutine 889 [running]:
k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher.(*Watcher).traversePluginDir.func1.1(0xc000d28720, 0xc001087640, 0x36)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/plugin_watcher.go:191 +0x68
created by k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher.(*Watcher).traversePluginDir.func1
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/plugin_watcher.go:190 +0x1ad
kubelet.service: Service hold-off time over, scheduling restart.
Stopped Kubernetes kubelet.
Started Kubernetes kubelet.
kubelet.service: Main process exited, code=exited, status=255/n/a
kubelet.service: Unit entered failed state.
kubelet.service: Failed with result 'exit-code'.
kubelet.service: Service hold-off time over, scheduling restart.
Stopped Kubernetes kubelet.
Started Kubernetes kubelet.
```

--- a/pkg/kubelet/util/pluginwatcher/plugin_watcher.go
+++ b/pkg/kubelet/util/pluginwatcher/plugin_watcher.go
@@ -187,7 +187,9 @@ func (w *Watcher) traversePluginDir(dir string) error {
 				return fmt.Errorf("failed to watch %s, err: %v", path, err)
 			}
 		case mode&os.ModeSocket != 0:
+			w.wg.Add(1)
 			go func() {
+				defer w.wg.Done()
 				w.fsWatcher.Events <- fsnotify.Event{
 					Name: path,
 					Op:   fsnotify.Create,

--- a/pkg/kubelet/util/pluginwatcher/plugin_watcher.go
+++ b/pkg/kubelet/util/pluginwatcher/plugin_watcher.go
@@ -176,9 +176,14 @@ func (w *Watcher) init() error {
 
 // Walks through the plugin directory discover any existing plugin sockets.
 // Goroutines started here will be waited for in Stop() before cleaning up.
+// Ignore all errors except root dir not being walkable
 func (w *Watcher) traversePluginDir(dir string) error {
 	return w.fs.Walk(dir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
+			if path == dir {
+				return fmt.Errorf("error accessing path: %s error: %v", path, err)
+			}
+
 			glog.Errorf("error accessing path: %s error: %v", path, err)
 			return nil
 		}

--- a/pkg/kubelet/util/pluginwatcher/plugin_watcher.go
+++ b/pkg/kubelet/util/pluginwatcher/plugin_watcher.go
@@ -175,10 +175,12 @@ func (w *Watcher) init() error {
 }
 
 // Walks through the plugin directory discover any existing plugin sockets.
+// Goroutines started here will be waited for in Stop() before cleaning up.
 func (w *Watcher) traversePluginDir(dir string) error {
 	return w.fs.Walk(dir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
-			return fmt.Errorf("error accessing path: %s error: %v", path, err)
+			glog.Errorf("error accessing path: %s error: %v", path, err)
+			return nil
 		}
 
 		switch mode := info.Mode(); {

